ValidationLibrary.Tests/RepositoryValidatorTests.cs

```diff
@@ -0,0 +1,80 @@
+
+namespace ValidationLibrary.Tests
+{
+    [TestFixture]
+    public class RepositoryValidatorTests
+    {
+        private const string ConfigFileName = "repository-validator.json";
+
+        private RepositoryValidator _repositoryValidator;
+
+        private IValidationRule[] _mockRules;
+        private IGitHubClient _mockClient;
+        //private IRepositoriesClient _mockRepositoriesClient;
+        private IRepositoryContentsClient _mockRepositoryContentsClient;
+
+        [SetUp]
+        public void SetUp()
+        {
+
+            var mockLogger = Substitute.For<ILogger<RepositoryValidator>>();
+
+            _mockClient = Substitute.For<IGitHubClient>();
+
+            _mockRules = new IValidationRule[]
+            {
+                Substitute.For<IValidationRule>(),
+                Substitute.For<IValidationRule>(),
+                Substitute.For<IValidationRule>()
+            };
+
+            _repositoryValidator = new RepositoryValidator(mockLogger, _mockClient, _mockRules);
+
+            //_mockRepositoriesClient = Substitute.For<IRepositoriesClient>();
+            _mockRepositoryContentsClient = Substitute.For<IRepositoryContentsClient>();
+            _mockClient.Repository.Content.Returns(_mockRepositoryContentsClient);
+        }
+
+        [Test]
+        public async Task Init_InitiatesAllRules()
+        {
+            await _repositoryValidator.Init();
+
+            foreach (var mockRule in _mockRules)
+            {
+                await mockRule.Received().Init(_mockClient);
+            }
+        }
+
+        [Test]
+        public async Task Validate_ValidatesWithAllRulesIfNoneAreIgnoredAndConfigIsNotFound()
+        {
+            _mockRepositoryContentsClient
+                .GetAllContents("testOwner", "mock-repositry", ConfigFileName)
+                .Returns<Task<IReadOnlyList<RepositoryContent>>>(x => { throw new NotFoundException("no message", HttpStatusCode.NotFound); });
+
+            var repository = CreateRepository("testOwner", "mock-repositry", true, false);
+            var result = await _repositoryValidator.Validate(repository, false);
+            Assert.NotNull(result);
+
+            foreach (var mockRule in _mockRules)
+            {
+                await mockRule.Received().IsValid(_mockClient, repository);
+            }
+        }
+
+        private Repository CreateRepository(string owner, string name, bool hasIssues, bool archived)
+        {
+            var user = new User(null, null, null, 0, null, DateTime.UtcNow, DateTime.UtcNow, 0, null, 0, 0, false, null, 0, 0, null, owner, null, null, 0, null, 0, 0, 0, null, new RepositoryPermissions(), false, null, null);
+            return new Repository(null, null, null, null, null, null, null, 0, null, user, name, $"{owner}/{name}", false, null, null, null, false, false, 0, 0, null, 0, null, DateTime.UtcNow, DateTime.UtcNow, null, null, null, null, hasIssues, false, false, false, 0, 0, null, null, null, archived, 0);
```

Comment:
That is just way too many individual constructor arguments. Are you sure the `Repository` is doing just one thing on one level of abstraction? Does the class have only one reason to change?

Reply:
I very much agree that there are too many arguments, but this class is provided by external library (Octokit) and is only used in tests.